Build failure at 7ab04b0 with gcc 14.2.1 on Fedora 41 x86_64

...

[openmohaa_7ab04b0_build_failure_gcc.txt](https://github.com/user-attachments/files/18466761/openmohaa_7ab04b0_build_failure_gcc.txt)

[238/702] Building C object code/sdl/CMakeFiles/omohsdl_client.dir/__/qcommon/common.c.o
FAILED: code/sdl/CMakeFiles/omohsdl_client.dir/__/qcommon/common.c.o 
/usr/bin/gcc -DAPP_MODULE -DNDEBUG -I/tmp/openmohaa/code/qcommon/../qcommon -I/tmp/openmohaa/.cmake/code/qcommon/generated -I/tmp/openmohaa/code/qcommon/../script -I/tmp/openmohaa/code/zlib -isystem /usr/include/SDL2 -Wno-comment -Werror=return-type -fno-gnu-unique -MD -MT code/sdl/CMakeFiles/omohsdl_client.dir/__/qcommon/common.c.o -MF code/sdl/CMakeFiles/omohsdl_client.dir/__/qcommon/common.c.o.d -o code/sdl/CMakeFiles/omohsdl_client.dir/__/qcommon/common.c.o -c /tmp/openmohaa/code/qcommon/common.c
/tmp/openmohaa/code/qcommon/common.c: In function ‘Com_Printf’:
/tmp/openmohaa/code/qcommon/common.c:314:53: error: passing argument 1 of ‘localtime_r’ from incompatible pointer type [-Wincompatible-pointer-types]
  314 |                                         localtime_r(&tms_local, &t);
      |                                                     ^~~~~~~~~~
      |                                                     |
      |                                                     struct tm *
In file included from /tmp/openmohaa/code/qcommon/q_shared.h:172,
                 from /tmp/openmohaa/code/qcommon/common.c:24:
/usr/include/time.h:159:57: note: expected ‘const time_t * restrict’ {aka ‘const long int * restrict’} but argument is of type ‘struct tm *’
  159 | extern struct tm *localtime_r (const time_t *__restrict __timer,
      |                                ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
/tmp/openmohaa/code/qcommon/common.c:314:65: error: passing argument 2 of ‘localtime_r’ from incompatible pointer type [-Wincompatible-pointer-types]
  314 |                                         localtime_r(&tms_local, &t);
      |                                                                 ^~
      |                                                                 |
      |                                                                 time_t * {aka long int *}
/usr/include/time.h:160:54: note: expected ‘struct tm * restrict’ but argument is of type ‘time_t *’ {aka ‘long int *’}
  160 |                                struct tm *__restrict __tp) __THROW;
      |                                ~~~~~~~~~~~~~~~~~~~~~~^~~~
/tmp/openmohaa/code/qcommon/common.c:315:50: error: passing argument 1 of ‘gmtime_r’ from incompatible pointer type [-Wincompatible-pointer-types]
  315 |                                         gmtime_r(&tms_gm, &t);
      |                                                  ^~~~~~~
      |                                                  |
      |                                                  struct tm *
/usr/include/time.h:154:54: note: expected ‘const time_t * restrict’ {aka ‘const long int * restrict’} but argument is of type ‘struct tm *’
  154 | extern struct tm *gmtime_r (const time_t *__restrict __timer,
      |                             ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
/tmp/openmohaa/code/qcommon/common.c:315:59: error: passing argument 2 of ‘gmtime_r’ from incompatible pointer type [-Wincompatible-pointer-types]
  315 |                                         gmtime_r(&tms_gm, &t);
      |                                                           ^~
      |                                                           |
      |                                                           time_t * {aka long int *}
/usr/include/time.h:155:51: note: expected ‘struct tm * restrict’ but argument is of type ‘time_t *’ {aka ‘long int *’}
  155 |                             struct tm *__restrict __tp) __THROW;
      |                             ~~~~~~~~~~~~~~~~~~~~~~^~~~
/tmp/openmohaa/code/qcommon/common.c: In function ‘Com_Error’:
/tmp/openmohaa/code/qcommon/common.c:525:1: warning: ‘noreturn’ function does return
  525 | }
      | ^
/tmp/openmohaa/code/qcommon/common.c: In function ‘Com_Quit_f’:
/tmp/openmohaa/code/qcommon/common.c:552:1: warning: ‘noreturn’ function does return
  552 | }
      | ^
[241/702] Building CXX object code/sdl/CMakeFiles/omohsdl_client.dir/__/qcommon/cm_trace_lbd.cpp.o
ninja: build stopped: subcommand failed.

--- a/code/qcommon/common.c
+++ b/code/qcommon/common.c
@@ -307,12 +307,9 @@ void QDECL Com_Printf( const char *fmt, ... ) {
 #ifdef WIN32
 					localtime_s(&tms_local, &t);
 					gmtime_s(&tms_gm, &t);
-#elif defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__)
-					localtime_r(&t, &tms_local);
-					gmtime_r(&t, &tms_gm);
 #else
-					localtime_r(&tms_local, &t);
-					gmtime_r(&tms_gm, &t);
+                    localtime_r(&t, &tms_local);
+                    gmtime_r(&t, &tms_gm);
 #endif
 					t_gmt = mktime(&tms_gm);
 					tz = difftime(t, t_gmt) / 60.0 / 60.0;

